python/python/lance/progress.py

```diff
@@ -126,14 +126,14 @@ def __init__(self, base_uri: str, metadata: Optional[Dict[str, str]] = None):
 
         fs, path = FileSystem.from_uri(base_uri)
         self._fs = fs
-        self._base_path = path
+        self._base_path: str = path
         self._metadata = metadata if metadata else {}
 
-    def _in_progress_path(self, fragment: "FragmentMetadata"):
-        return self._base_path / f"fragment_{fragment.id}.in_progress"
+    def _in_progress_path(self, fragment: "FragmentMetadata") -> str:
+        return self._base_path + f"/fragment_{fragment.id}.in_progress"
```

Comment:
is `_base_path` a `pathlib.Path`? If so, it is recommended to use `path / path` to build a file path.

Reply:
It is a string, but if you really want we can make it a Path.

Reply:
Tried it, and the PyArrow filesystem API doesn't accept paths. So I'm not sure it's worth it to convert it to a path if we have to immediately convert back. Since this might be on object storage, we don't want any kind of windows path handling here.